api/test/utils/reader.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"errors"
+	"io"
+)
+
+type TestReader struct {
+	data         []byte
+	allowedLoops int
+	loopCount    int
+	pos          int
+	throwError   bool
+}
+
+func NewTestReader(data []byte, allowedLoops int, throwError bool) *TestReader {
+	return &TestReader{
+		data:         data,
+		allowedLoops: allowedLoops,
+		throwError:   throwError,
+	}
+}
```

Comment:
Hope i'm not nit-picking here... But I think this could benefit from some documentation, it's not immediately clear to me what this does, until I look up how it is used and why it is used in the actual test.

Reply:
It's a simple io.Reader implementation that goes boom on command to test that the code can handle I/O errors. I will add some comments...

Reply:
Yep I got that too, after looking at the test were it was used. But when looking just at this single file, I didn't get what it can be used for (maybe that's the case for many test helpers...)